packages/react-components/react-dialog/src/utils/useDisableBodyScroll.ts

```diff
@@ -0,0 +1,55 @@
+import { useFluent_unstable } from '@fluentui/react-shared-contexts';
+import { useCallback } from 'react';
+
+const disableScrollElementProp = '__fluentDisableScrollElement' as const;
+
+type FluentDisableScrollElement = HTMLElement & {
+  [disableScrollElementProp]: {
+    count: number;
+    previousOverflowStyle: string;
+    previousPaddingRightStyle: string;
+  };
+};
+
+/**
+ * hook that disables body scrolling through `overflow: hidden` CSS property
+ */
+export function useDisableBodyScroll() {
+  const { targetDocument } = useFluent_unstable();
+  return useCallback(() => {
+    if (targetDocument) {
+      return disableScroll(targetDocument.body);
+    }
+  }, [targetDocument]);
+}
+
+/**
+ * disables scrolling from a given element through `overflow: hidden` CSS property
+ * @param target - element to disable scrolling from
+ * @returns a method for enabling scrolling again
+ */
+export function disableScroll(target: HTMLElement) {
+  const { clientWidth } = target.ownerDocument.documentElement;
+  const innerWidth = target.ownerDocument.defaultView?.innerWidth ?? 0;
+  assertIsDisableScrollElement(target);
+  if (target[disableScrollElementProp].count === 0) {
+    target.style.overflow = 'hidden';
+    target.style.paddingRight = `${innerWidth - clientWidth}px`;
```

Comment:
FYI: This assumes that `body` has no `padding-right` value, in the case `body` has `padding-right` then this might actually end up creating a jump effect anyway.

Reply:
The ideal solution would be to get the computed property `padding-right` and then sum it up with `innerWidth` and `clientWidth`